shared/src/commonMain/kotlin/io/customer/shared/tracking/api/model/BatchTracking.kt

```diff
@@ -0,0 +1,36 @@
+package io.customer.shared.tracking.api.model
+
+import io.customer.shared.tracking.model.TrackingError
+import kotlinx.serialization.SerialName
+
+/**
+ * Request class only to be used when making tracking requests in batch.
+ */
+@kotlinx.serialization.Serializable
+internal data class BatchTrackingRequestBody(
+    @SerialName("batch") val batch: List<TrackingRequest>,
+)
+
+/**
+ * Response class only to be used when making tracking requests in batch.
+ */
+@kotlinx.serialization.Serializable
+internal data class BatchTrackingResponseBody(
+    @SerialName("meta") val meta: BatchTrackingResponseMeta?,
+    @SerialName("errors") val errors: List<TrackingError>?,
+)
```

Comment:
Instead of having this data class contain 2 optionals that we will need to cast later on as non-null, can we do this instead? 

1. Perform the HTTP network request. 
2. Get response back. Check the returned status code to see if the request was successful. 
3. If successful, deserialize the response JSON body to an object like this:
```suggestion
internal data class BatchTrackingSuccessfulResponseBody(
    val meta: BatchTrackingResponseMeta
)
```
4. If unsuccessful, deserialize the response JSON body to an object like this:
```suggestion
internal data class BatchTrackingErrorResponseBody(
    val errors: List<TrackingError>,
)
```

Reply:
I need to test this first. But we are [not parsing it manually](https://github.com/customerio/mobile-shared/pull/4/files#diff-eccbabe54c4115c798e9ba72eda74c3ac8de8546434a088d248304d3eee7e8ceR47). The change you are suggesting will force us not to rely on ktor for this and get raw response instead, which is not ideal in my opinion for this case.

Also, I now realize that we are not using `BatchTrackingResponseMeta`. I will try to reproduce the error to get `meta` in response and confirm if we really don't need it or I just missed. If not, I'll remove this. But `errors` will remain this way as we don't get any errors in complete successful responses. It is only used for partially or complete incorrect tasks.

If you still think we should do this, can you share the idea why would you like to have it non-null? As I think we already have handled [errors list very cleanly](https://github.com/customerio/mobile-shared/pull/4/files#diff-eccbabe54c4115c798e9ba72eda74c3ac8de8546434a088d248304d3eee7e8ceR50) and the caller gets this in [BatchTrackingResponse](https://github.com/customerio/mobile-shared/pull/4/files) which is not null.

Reply:
I tried with invalid credentials (`siteid` and `apikey`) and received following error.

```
{
  "meta": {
    "error": "Unauthorized request"
  }
}
```

For now, I have [added logs for this](https://github.com/customerio/mobile-shared/pull/4/commits/e5f0b7c1fc87cc5af37955efbef7786d438abd5c) and added as action item to discuss this with Jatin and use it in a better way. As there could be more cases where we might want to inform the customers through logs to suggest what they can do to fix it.

Reply:
Oh, so `BatchTrackingResponseBody`'s job is only for JSON decoding? I am OK with having 2 optionals if that's the only purpose of it. 

I wanted to avoid this class being used in a lot of the logic of the code base. If we decode JSON into a `BatchTrackingResponseBody` instance, parse the `BatchTrackingResponseBody` object, and convert it into data classes like `BatchTrackingResponse` without optionals then that works.